src/os/keyvaluestore/KeyValueStore.cc

```diff
@@ -1213,9 +1213,8 @@ int KeyValueStore::_do_transactions(list<Transaction*> &tls, uint64_t op_seq,
   for (list<Transaction*>::iterator p = tls.begin();
        p != tls.end();
        ++p, trans_num++) {
-    r = _do_transaction(**p, bt, handle);
-    if (r < 0)
-      break;
+    unsigned ret = _do_transaction(**p, bt, handle);
+    assert(ret == 0);
```

Comment:
I don't necessarily agree with making this an assert().  Either _do_transaction() should have a void return or it could potentially return an error in the future,  So we should leave the code the way it was.